datadog_checks_base/datadog_checks/base/log.py

```diff
@@ -16,8 +16,15 @@
 # Arbitrary number less than 10 (DEBUG)
 TRACE_LEVEL = 7
 
+Logger = logging.Logger
 
-class AgentLogger(logging.getLoggerClass()):
+# Weird flex, but static type checkers need a statically-defined base class, so
+# we must assign the dynamic logger class in a type-ignored branch.
+if True:
+    Logger = logging.getLoggerClass()  # type: ignore
```

Comment:
Eww haha. Please tell me there's a way around this

Reply:
Unfortunately, there isn't: https://github.com/python/mypy/issues/2477#issuecomment-327198257

> *anything* with an expression that's not a simple class reference will cause that error

The recommended solution seems to be to use `typing.TYPE_CHECKING`, which might feel a bit cleaner? https://github.com/python/mypy/issues/2477#issuecomment-313984522

```python
import typing

def make_base():
    pass

if typing.TYPE_CHECKING:
    base = object
else:
    base = make_base()
```

So, for this particular case:

```python
import typing
import logging

if typing.TYPE_CHECKING:
    Logger = logging.Logger
else:
    Logger = logging.getLoggerClass()
```

(At least that way, there's no need for `# type: ignore`.)

Reply:
Yeah sure